lk2nd/smp/cpu-boot.c

```diff
@@ -56,6 +98,70 @@ int qcom_set_boot_addr(uint32_t addr, bool arm64)
 				QCOM_SCM_BOOT_FLAG_COLD_ALL, addr);
 }
 
+void qcom_power_up_l2_cache(uint32_t mpidr, uint32_t base)
+{
+	uint32_t val;
+
+	/* Skip if cluster L2 is already powered on */
+	val = readl(base + APCS_L2_PWR_STATUS);
+	if (val & APCS_L2_PWR_STATUS_L2_HS_STS) {
+		dprintf(INFO, "CPU (%d) L2 cache already powered-up\n", mpidr);
+		return;
+	}
+
+	/* Close L2/SCU Logic GDHS and power up the cache */
+	val = APCS_L2_PWR_CTL_L2_HS_CNT(16) |
+	      APCS_L2_PWR_CTL_L2_ARRAY_HS_CLAMP |
+	      APCS_L2_PWR_CTL_SCU_ARRAY_HS_CLAMP | APCS_L2_PWR_CTL_SYS_RESET |
+	      APCS_L2_PWR_CTL_L2_HS_RST | APCS_L2_PWR_CTL_L2_HS_EN |
+	      APCS_L2_PWR_CTL_L2_HS_CLAMP;
```

Comment:
Smallest nit: IMO it's a bit easier to read bits here if they're all aligned on the new line, so the `APCS_L2_PWR_CTL_` prefix is in the same place:

```suggestion
	/* Close L2/SCU Logic GDHS and power up the cache */
	val = APCS_L2_PWR_CTL_L2_HS_CNT(16) |
	      APCS_L2_PWR_CTL_L2_ARRAY_HS_CLAMP |
	      APCS_L2_PWR_CTL_SCU_ARRAY_HS_CLAMP |
	      APCS_L2_PWR_CTL_SYS_RESET |
	      APCS_L2_PWR_CTL_L2_HS_RST |
	      APCS_L2_PWR_CTL_L2_HS_EN |
	      APCS_L2_PWR_CTL_L2_HS_CLAMP;
```